### PR TITLE
[run_glue] Add MNLI compatible mode

### PR DESCRIPTION
**Pull request by JetRunner**
In this PR:

- Upgrade `datasets` to `1.3.0`
- Rename `datasets` variable to `task_datasets` in `run_glue.py` to avoid confusion with the library `datasets`
- Add a `--mnli_compat_mode` option to use the old label assignment for MNLI

### CODE REVIEW

**Comment by JetRunner** [comment]
The CI failed but it seems irrelevant. Could you please give it a check? @LysandreJik

File: examples/text-classification/run_glue.py [hunk 1]
```diff
@@ -207,7 +215,25 @@ def main():
...
         # Downloading and loading a dataset from the hub.
-        datasets = load_dataset("glue", data_args.task_name)
+        if data_args.task_name == "mnli":
+            if data_args.mnli_compat_mode:
+                logger.warning(
+                    "The labels assignment is overwritten with the old assignment"
+                    "['contradiction', 'entailment', 'neutral'] from transformers<3.5.0."
+                )
+                if version.parse(datasets.__version__) < version.parse("1.3.0"):
+                    raise ImportError("Please update `datasets` to >1.3.0 to overwrite the labels.")
```
Comment (patil-suraj) [inline examples/text-classification/run_glue.py:225]:
```suggestion
                    raise ImportError("Please update `datasets` to >=1.3.0 to overwrite the labels.")
```

**Review by patil-suraj — APPROVED** [review]
LGMT! I'm okay with `datasets` => `task_datasets`

**Comment by patil-suraj** [comment]
The CI issue is `test_hf_api`, which is fixed on master now, rebasing should make the CI green!

**Review by lhoestq — APPROVED** [review]

**Review by sgugger — COMMENTED** [review]
I really dislike the change as it makes the script (which we showcase as an example for beginners) more complicated and less understandable. We can add a script instead that will add the proper label assignment in the model config, which this example will then use if available and put the warning in the README instead.

**Comment by JetRunner** [comment]
@sgugger Sorry but I can't really agree with you here. It is a problem introduced by mistake and we shouldn't just try to ignore that. It just adds a few lines of if-else and I don't think it'll add too much burden for users. 

I am never a fan of trainer-style code since IMO capsulation is not simplicity (I believe our user survey supports my claim here). Besides, I do think back-compatibility matters especially when our success depends heavily on the prosperity of the model hub. This PR is meant to fix a bug though it's harsh. Happy to have more discussion here.

**Comment by sgugger** [comment]
So we discussed it a bit more internally. This particular example script will stay as is with the backward-compatibility problem (which isn't one IMO since the problem is in the model config not having the right labels). As I said before it's an example aimed at data scientists that shouldn't necessarily have all the functionality.

There will be another script for GLUE (probably by the end of the month) very soon that doesn't use `Trainer` and has the training loop exposed, where we can integrate your fix.

**Comment by JetRunner** [comment]
Looking forward to the new script you mentioned!

File: examples/text-classification/run_glue.py [hunk 2]
```diff
@@ -207,7 +215,25 @@ def main():
...
+                task_datasets = datasets.load_dataset(
+                    "glue", data_args.task_name, label_classes=["contradiction", "entailment", "neutral"]
```
Comment (AliHadizadeh) [inline examples/text-classification/run_glue.py:234]:
```suggestion
                    "glue", data_args.task_name, label_classes=["contradiction", "neutral" , "entailment"]
```
I believe this is the correct order!

**Review by AliHadizadeh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**